apps/server/src/modules/room-membership/service/room-membership.service.spec.ts

```diff
@@ -318,6 +319,97 @@ describe('RoomMembershipService', () => {
 		});
 	});
 
+	describe('changeRoleOfRoomMembers', () => {
+		describe('when roomMembership does not exist', () => {
+			it('should throw an exception', async () => {
+				roomMembershipRepo.findByRoomId.mockResolvedValue(null);
+
+				await expect(
+					service.changeRoleOfRoomMembers(new ObjectId().toHexString(), [], RoleName.ROOMEDITOR)
+				).rejects.toThrowError(BadRequestException);
+			});
+		});
+
+		describe('when roomMembership exists', () => {
+			const setup = () => {
+				const user = userFactory.buildWithId();
+				const otherUser = userFactory.buildWithId();
+				const userNotInRoom = userFactory.buildWithId();
+				const school = schoolFactory.build();
+				const viewerRole = roleFactory.buildWithId({ name: RoleName.ROOMVIEWER });
+				const editorRole = roleFactory.buildWithId({ name: RoleName.ROOMEDITOR });
+				const group = groupFactory.build({
+					type: GroupTypes.ROOM,
+					organizationId: school.id,
+					users: [
+						{ userId: user.id, roleId: viewerRole.id },
+						{ userId: otherUser.id, roleId: viewerRole.id },
+					],
+				});
+				const room = roomFactory.build({ schoolId: school.id });
+				const roomMembership = roomMembershipFactory.build({
+					roomId: room.id,
+					userGroupId: group.id,
+					schoolId: school.id,
+				});
+
+				roomMembershipRepo.findByRoomId.mockResolvedValue(roomMembership);
+				groupService.findById.mockResolvedValue(group);
+				roleService.findByName.mockResolvedValue(editorRole);
+
+				return {
+					user,
+					otherUser,
+					userNotInRoom,
+					room,
+					roomMembership,
+					group,
+					viewerRole,
+					editorRole,
+				};
+			};
+
+			it('should change role of user to editor', async () => {
+				const { user, room, group, editorRole } = setup();
+
+				await service.changeRoleOfRoomMembers(room.id, [user.id], RoleName.ROOMEDITOR);
+
+				expect(groupService.save).toHaveBeenCalledWith(
+					expect.objectContaining({
+						id: group.id,
+						users: expect.arrayContaining([{ userId: user.id, roleId: editorRole.id }]) as GroupUser[],
+					})
+				);
+			});
+
+			it('should not change role of other user', async () => {
```

Comment:
What is the difference between user and otherUser that leads to this expectation?
Can't see it...

Reply:
the difference is that otherUser is not in the list of ids that are passed to the endpoint to change :)